Mixpanel/MixpanelInstance.swift

```diff
@@ -845,19 +850,42 @@ extension MixpanelInstance {
      - parameter completion: an optional completion handler for when the flush has completed.
      */
     open func flush(completion: (() -> Void)? = nil) {
-        serialQueue.async() {
+        
+        self.archive()
+
+        networkQueue.async() {
+            
             if let shouldFlush = self.delegate?.mixpanelWillFlush(self), !shouldFlush {
                 return
             }
+            
+            self.trackingQueue.sync {
+                self.copyEventsQueue = self.eventsQueue
+                self.copyPeopleQueue = self.people.peopleQueue
+
+                self.eventsQueue.removeAll()
```

Comment:
could there be a case where eventsQueue or peopleQueue are written or read during this time?